src/CommandHandler/Wishlist/ExportSelectedProductsFromWishlistToPdfHandler.php

```diff
@@ -0,0 +1,124 @@
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\ExportSelectedProductsFromWishlistToPdfInterface;
+use BitBag\SyliusWishlistPlugin\Model\Factory\VariantPdfModelFactoryInterface;
+use BitBag\SyliusWishlistPlugin\Resolver\VariantImagePathResolverInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Dompdf\Dompdf;
+use Dompdf\Options;
+use phpDocumentor\Reflection\Types\This;
+use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Environment;
+
+final class ExportSelectedProductsFromWishlistToPdfHandler implements MessageHandlerInterface
+{
+    private ProductVariantRepositoryInterface $productVariantRepository;
+
+    private VariantImagePathResolverInterface $variantImagePathResolver;
+
+    private VariantPdfModelFactoryInterface $variantPdfModelFactory;
+
+    private Environment $twigEnvironment;
+
+    private bool $isSelected = false;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private RequestStack $request;
+
+    public function __construct(
+        ProductVariantRepositoryInterface $productVariantRepository,
+        VariantImagePathResolverInterface $variantImagePathResolver,
+        VariantPdfModelFactoryInterface $variantPdfModelFactory,
+        Environment $twigEnvironment,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator,
+        RequestStack $request
+    ) {
+        $this->productVariantRepository = $productVariantRepository;
+        $this->variantImagePathResolver = $variantImagePathResolver;
+        $this->variantPdfModelFactory = $variantPdfModelFactory;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+        $this->request = $request;
+    }
+
+    public function __invoke(ExportSelectedProductsFromWishlistToPdfInterface $exportSelectedProductsFromWishlistToPdf): void
+    {
+        $wishlistProducts = $exportSelectedProductsFromWishlistToPdf->getWishlistProducts();
+        $selectedProductsModel = $this->createModelToPdf($wishlistProducts,$this->request->getCurrentRequest());
+        $this->exportToPdf($selectedProductsModel);
+    }
+
+    private function createModelToPdf(Collection $wishlistProducts, Request $request): array
+    {
+        $selectedProducts = [];
+
+        /** @var AddWishlistProductInterface $wishlistProduct */
+        foreach ($wishlistProducts as $wishlistProduct) {
+            if ($wishlistProduct->isSelected()) {
```

Comment:
1. https://dev.to/tadeubdev/object-calisthenics-with-php-en-2h4p — it will decrease amount of indented code significantly
2. break this into separate private methods or — even better — to separate service that will take care of generating the model.